src/PhpSpec/Formatter/TapFormatter.php

```diff
@@ -0,0 +1,136 @@
+{
+
+    const VERSION = 'TAP version 13';
+
+    const OK = 'ok %d';
+
+    const NOT_OK = 'not ok %d';
+
+    const DESC = ' - %s: %s';
+
+    const SKIP = ' # SKIP %s';
+
+    const TODO = ' # TODO %s';
+
+    const PLAN = '1..%d';
+
+    const MESSAGE = "---\nmessage: %s\n...";
+
+    /**
+     * @var int
+     */
+    private $examplesCount = 0;
+
+    /**
+     * @var string
+     */
+    private $currentSpecificationTitle;
+
+    /**
+     * @param SuiteEvent $event
+     */
+    public function beforeSuite(SuiteEvent $event)
+    {
+        $this->getIO()->writeln(self::VERSION);
+    }
+
+    /**
+     * @param SpecificationEvent $event
+     */
+    public function beforeSpecification(SpecificationEvent $event)
+    {
+        $this->currentSpecificationTitle = $event->getSpecification()->getTitle();
+    }
+
+    /**
+     * @param ExampleEvent $event
+     */
+    public function afterExample(ExampleEvent $event)
+    {
+        $this->examplesCount++;
+        $desc = sprintf(
+            self::DESC,
+            $this->currentSpecificationTitle,
+            preg_replace('/^its? /', '', $event->getExample()->getTitle())
+        );
+
+        switch ($event->getResult()) {
+            case ExampleEvent::PASSED:
+                $result = sprintf(self::OK, $this->examplesCount) . $desc;
+                break;
+            case ExampleEvent::PENDING:
+                $message = sprintf(self::TODO, $this->getExceptionMessage($event));
+                $result = sprintf(self::OK, $this->examplesCount) . $desc . $message;
+                break;
+            case ExampleEvent::SKIPPED:
+                $message = sprintf(self::SKIP, $this->getExceptionMessage($event));
+                $result = sprintf(self::OK, $this->examplesCount) . $desc . $message;
+                break;
+            case ExampleEvent::BROKEN:
+            case ExampleEvent::FAILED:
+                $message = $this->getExceptionMessage($event, true);
+                $result = sprintf(self::NOT_OK, $this->examplesCount) . $desc . "\n" . $message;
+                break;
+        }
+
+        $this->getIO()->writeln($result);
+    }
+
+    /**
+     * @param SuiteEvent $event
+     */
+    public function afterSuite(SuiteEvent $event)
+    {
+        $this->getIO()->writeln(sprintf(
+            self::PLAN,
+            $this->getStatisticsCollector()->getEventsCount()
+        ));
+    }
+
+    /**
+     * Format message as two-space indented YAML when needed outside of a
+     * SKIP or TODO directive.
+     *
+     * @param ExampleEvent $event
+     * @param boolean $yaml
+     * @return string
+     */
+    private function getExceptionMessage(ExampleEvent $event, $yaml = false)
+    {
+        if (false === $yaml) {
+            return str_replace(
+                array("\r\n", "\n", "\r"),
+                ' \\ ',
+                $event->getException()->getMessage()
+            );
+        }
+        $message = $event->getException()->getMessage();
```

Comment:
This has shifted around a bit in last commit. Probably a few other pieces of cleanup, but this is close to done unless I start digging into data blocks. We'll see!